spec/support/system_helpers.rb

```diff
@@ -138,4 +138,25 @@ def select_text_range(selector, start = 0, offset = 5)
 
     page.execute_script(js, selector, start, offset)
   end
+
+  def setup_s3_system_test
+    SiteSetting.enable_s3_uploads = true
+
+    SiteSetting.s3_upload_bucket = "discoursetest"
+    SiteSetting.enable_upload_debug_mode = true
+
+    SiteSetting.s3_access_key_id = MinioRunner.config.minio_root_user
+    SiteSetting.s3_secret_access_key = MinioRunner.config.minio_root_password
+    SiteSetting.s3_endpoint = MinioRunner.config.minio_server_url
+
+    MinioRunner.start
```

Comment:
From trying to get this running locally, I think it would be great if we could add some sanity checks here with some automatic help text.

e.g. it could try making a request to `minio.local` to make sure you have the `/etc/hosts` setup correctly. And perhaps it could make sure the response time is sub-4-seconds to detect the macOS `.local` issue?

Reply:
@davidtaylorhq I actually do a health check in the runner since we need to check that the server is live before sending it `mc` calls; but I agree it could be improved to show a better error and detect the macOS local if a) the platform is macOS and b) the domain ends with .local . Will do that in the gem :+1: